spec/models/book_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+describe Book do
+  subject(:book) { build(:book) }
+
+  it { is_expected.to be_valid }
+  it { is_expected.to validate_presence_of(:genre) }
+  it { is_expected.to validate_presence_of(:author) }
+  it { is_expected.to validate_presence_of(:image) }
+  it { is_expected.to validate_presence_of(:title) }
+  it { is_expected.to validate_presence_of(:editor) }
+  it { is_expected.to validate_presence_of(:year) }
+  it do
+    expect(book.year).to satisfy do |v|
```

Comment:
why you didn't use the the numericality validate method ? 🤔  
something like this `validate_numericality_of(:year).only_integer`